core/src/main/scala/akka/kafka/KafkaConsumerActor.scala

```diff
@@ -54,23 +54,18 @@ object KafkaConsumerActor {
     private[KafkaConsumerActor] class NoPollResult extends RuntimeException with NoStackTrace
   }
 
-  private[kafka] def rebalanceListener(onAssign: Iterable[TopicPartition] => Unit, onRevoke: Iterable[TopicPartition] => Unit) = new ConsumerRebalanceListener {
-    override def onPartitionsAssigned(partitions: util.Collection[TopicPartition]): Unit = {
-      onAssign(partitions.asScala)
-    }
-    override def onPartitionsRevoked(partitions: util.Collection[TopicPartition]): Unit = {
-      onRevoke(partitions.asScala)
-    }
-  }
+  private[kafka] case class ListenerCallbacks(onAssign: (KafkaConsumer[_, _], Iterable[TopicPartition]) => Unit, onRevoke: Iterable[TopicPartition] => Unit)
```

Comment:
scala.Iterable might be mutable, would feel safer to use something that for sure is immutable, such as scala.collection.immutable.Seq, or `Set`. I think it's converted to `Set` later anyway.